### ISSUE TITLE
Automatic scaling for autograd.grad

### ISSUE THREAD

**Issue by Aixile** [NVIDIA/apex#873]
When using double backpropagation to get second-order derivatives, if gradient overflow happens in the first backprop, the result of `autograd.grad` would be `None`, so the model will never recover.
I wonder is there any way to handle the scaling factor for the `outputs` parameter of `autograd.grad` automatically?

**Comment by mcarilli** [comment]
This isn't possible with apex amp.  However, it is supported with Pytorch native amp:
https://pytorch.org/docs/master/amp.html
which I intend to replace apex amp.  See
https://pytorch.org/docs/master/notes/amp_examples.html#gradient-penalty

Native amp is not fully available in 1.5, but is fully available now in pytorch master and nightly packages, and will be in the 1.6 release.

**Comment by Aixile** [comment]
Thank you, I would like to try it!
